packages/types/src/transaction.ts

```diff
@@ -63,6 +63,11 @@ export interface Transaction extends TransactionContext, Span {
    */
   data: { [key: string]: any };
 
+  /**
+   * Metadata about the transaction
+   */
+  metadata: TransactionMetadata;
+
```

Comment:
> why did we add this ?

Because we now need to access it from within the `Span` class. Since the `Transaction` class extends the `Span` class, it imports from `'./span'`. If the `Span` class also imported the `Transaction` class (from `'./transaction'`), we'd end up in an import loop. `Span` therefore has to import `Transaction` instead from from `@sentry/types`, which means the type needs to know about the `metadata` property.

> I can't quite figure out why we pass tracestate since we can figure it out from the headers

This _is_ us having figured it out from the headers. The headers are passed to `extractTracestateData`, and the data then comes through `startTransaction` to the constructor (for example [here](https://github.com/getsentry/sentry-javascript/blob/2a7b830938aeaef8acc66a3b5b4a902e34e79df6/packages/node/src/handlers.ts#L63-L75).)